docs/DeveloperGuide.md

```diff
@@ -286,31 +342,183 @@ Use case ends.
+
+2a. The list is empty
+
+> Use case ends
+
+3a. The given index is invalid
+
+> 3a1. System shows an error message (“Please select a task on the list with a valid index”)
+> Use case resumes at step 2
 
-1. Should work on any [mainstream OS](#mainstream-os) as long as it has Java `1.8.0_60` or higher installed.
-2. Should be able to hold up to 1000 persons.
-3. Should come with automated unit tests and open source code.
-4. Should favor DOS style commands over Unix-style commands.
+3b. The new input time format is invalid
 
-{More to be added}
+> 3b1. System shows an error message (“Please follow the given time format”)
+> Use case resumes at step 2
+
+### Use case 05 - Undo the previous command that modified the task list
+
+**MSS**
+
+1. Actor enters an undo command
+2. System finds the latest command that modified the task list
+3. System undo the identified command
+4. System shows a feedback message (“The command <last-command> has been undone”) and displays the updated list on the interface.
+5. Use case ends.
+
+**Extensions**
+
+2a. There are no previous commands that modify the list (since the launch of the application)
+
+> 2a1. System shows an error message (“No previous command to undo”)
+> Use case ends
+
+### Use case 06 - Mark a task as completed
+
+**MSS**:
+
+1. Actor requests to list tasks
+2. System show a list of tasks
+3. Actor requests to mark a task specified by its index in the list as completed
+4. System updates the task
+5. System shows a feedback message (“Task <index> is marked as completed”) and hides the marked task.
+6. Use case ends
+
+**Extensions**
+
+2a. The list is empty
+
+> 2a1. Use case ends
+
+3a. The given index is invalid
+
+> 3a1. System shows an error message (“Please select a task on the list with a valid index”)
+> Use case resumes at step 2
+
+### Use case 07 - Add short hand commands
+
+**MSS**
+
+1. Actor enters a alias command and specify the name and new alias name of the command
+2. System alias the command
+3. System shows a feedback message (“The command <original-command> can now be keyed in as <new-command>”)
+4. Use case ends.
+
+**Extensions**
+
+1a. There is no existing command with the original name specified
+
+> 1a1. System shows an error message (“There is no such existing command”)
+> Use case ends
+
+1b. The new alias name is already reserved/used for other commands
+
+> 1b1. System shows an error message (“The name is already in use”)
+> Use case ends
+
+*a. At any time, Actor choose to exit System
+
+> *a1. System displays a goodbye message
+> *a2. System closes the program
+
+*b. At any time, Actor enters a invalid command
+
+> *b1.  System gives an error message (“We do not understand the command: <invalid-command>”)
+> *b2. System displays a short list of valid commands
+
+
+## Appendix C : Non Functional Requirements
+
+1.  Should work on any [mainstream OS](#mainstream-os) as long as it has Java `1.8.0_60` or higher installed.
+2.	Should be able to hold up to 500 tasks.
```

Comment:
500 tasks per user? 500 live tasks or does that include tasks that have been completed, marked and archived?